resdk/shortcuts/collection.py

```diff
@@ -0,0 +1,450 @@
+            flattened = []
+
+            for relation in relations:
+                if any(obj['position'] for obj in relation.entities):
+                    flattened.append({
+                        rel['position']: sample_id_to_slug(rel['entity'])
+                        for rel in relation.entities
+                    })
+                    flattened[-1]['_id'] = relation.id
+                else:
+                    flattened.append({
+                        'samples': [
+                            sample_id_to_slug(rel['entity']) for rel in relation.entities
+                        ],
+                        '_id': relation.id,
+                    })
+
+            return flattened
+
+        def add_default_values(relations):
+            """Add default values for missing keys."""
+            if 'compare' not in relations:
+                relations['compare'] = {
+                    'background': [
+                        {'sample': '', 'background': ''},
+                    ],
+                    'case-control': [
+                        {'case': '', 'control': ''},
+                    ],
+                }
+
+            if 'group' not in relations:
+                relations['group'] = {
+                    'replicates': [
+                        {'samples': ['']},
+                    ],
+                }
+
+            if 'series' not in relations:
+                relations['series'] = {
+                    'time-series': [
+                        {'<time_0>': '', '<time_1>': ''},
+                    ],
+                    'drug-dose': [
+                        {'<dose_0>': '', '<dose_1>': ''},
+                    ],
+                }
+
+            return relations
+
+        default_name = '{}_relations.yml'.format(self.slug)
+        if path is None:
+            path = os.path.join(os.getcwd(), default_name)
+        elif os.path.isdir(path):
+            path = os.path.join(path, default_name)
+
+        # Double default dict for sorting relations. First level is
+        # relation type and second level is relation label (or None if
+        # label is not set).
+        relations = defaultdict(lambda: defaultdict(list))
+
+        for rel in self.resolwe.relation.filter(collection=self.id):
+            relations[rel.type][rel.label].append(rel)
+
+        relations = dict(relations)
+        for rel_type in relations.keys():
+            labels = list(relations[rel_type].keys())
+
+            # Strip label level if no label is present.
+            if labels == [None]:
+                relations[rel_type] = flatten_relations(relations[rel_type][None])
+            else:
+                relations[rel_type] = {
+                    key: flatten_relations(value)
+                    for key, value in six.iteritems(relations[rel_type])
+                }
+
+        relations = add_default_values(relations)
+
+        help_text = '''\
```

Comment:
Perhaps this should be moved into a separate file, containing just the template, as this interrupts the code flow?